DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,113 @@
+# Developer guide
+
+This document describes how to set up a development environment for Calico, as well as how to build and test development code.
+
+## Requirements
+
+These build instructions assume you have a Linux build environment
+with:
+
+-  Docker 1.12+
+-  git
+-  make
+
+## Checking out the code
+
+Calico code is distributed across several different components, each of which lives in its own git repository. To build
+all of Calico, you will need to check out each repository into a common base directory (for example, `$GOPATH/src/github.com/projectcalico`).
+
+You will need the following repositories cloned on your development machine:
+
+```
+BASEDIR
+└───calico
+└───libcalico-go
+└───confd
```

Comment:
Should this be at kelseyhightower/?  I think glide complains if nothing else

Reply:
glide might emit a warning, that's true.

FWIW this is how I have my env set up and it works just fine. I think since all the builds are containerized and we use glide to alias it as `kelseyhightower` we're probably OK. I like a common base directory since it's easier to reason about.

Longer term, I'd like to just ditch confd altogether (it doesn't even resemble upstream any more) and just move what we need into the `node` repo, but obvs that's way outside the scope of this PR :D

Reply:
> I think since all the builds are containerized and we use glide to alias it as kelseyhightower we're probably OK

I haven't had to go through this set up yet (specifically the `confd` --> `kelseyhightower` dependency), but would it be worth explicitly mentioning the glide alias here? Or is that something that happens in our make target (i.e. we don't have to do that manually)?

Reply:
@stevegaossou it's checked into the repo, so there's no work required on the user.

See here: https://github.com/projectcalico/node/blob/master/glide.yaml#L9-L15